nifi-nar-bundles/nifi-site-to-site-reporting-bundle/nifi-site-to-site-reporting-task/src/main/java/org/apache/nifi/reporting/SiteToSiteBulletinReportingTask.java

```diff
@@ -153,8 +164,12 @@ public void onTrigger(final ReportingContext context) {
             attributes.put("reporting.task.type", this.getClass().getSimpleName());
             attributes.put("mime.type", "application/json");
 
-            final byte[] data = jsonArray.toString().getBytes(StandardCharsets.UTF_8);
-            transaction.send(data, attributes);
+            if(context.getProperty(RECORD_WRITER).isSet()) {
```

Comment:
The first one you did also has a customValidate method, can we put that logic into the base class (not necessarily in the exact customValidate() method but something all of these can call)? If we add the property to the base class, then it could be in the base class's customValidate(), your call :)